nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,189 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  defaultUser = "webhook";
+
+  hooksFile = pkgs.writeText "webhook-config" (
+    builtins.toJSON (
+      builtins.attrValues cfg.hooks
+    )
+  );
+
+  # Todo: more expressive type?
+  typeTriggerRule = types.attrs;
+
+in {
+  options = {
+    services.webhook = {
+      enable = mkEnableOption (mdDoc "Webhook is server written in Go, that allows you to create HTTP endpoints (hooks), which execute configured commands for any person or service that knows the URL.");
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.webhook;
+        defaultText = "pkgs.webhook";
+        example = literalExample "pkgs.webhook";
+        description = mdDoc ''
+          Webhook package to use.
+        '';
+      };
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = mdDoc ''
+          Webhook will be run under this user (user will be created if it doesn't exist. This can be your user name).
+        '';
+      };
+      group = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = mdDoc ''
+          Webhook will be run under this group (group will be created if it doesn't exist. This can be your group name).
+        '';
+      };
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
```

Comment:
This binds on all interfaces, but there's no `openFirewall` option.
Should `webhook` be locally reverse proxied or not?

Reply:
Added an `openFirewall` option in https://github.com/NixOS/nixpkgs/pull/205636/commits/c5501723ef2ab062bccc7cdb8ecd433fc549434b

It'll be up to the user to decide whether they want to reverse proxy it or not